pandas/__init__.py

```diff
@@ -118,6 +117,26 @@
 __git_version__ = v.get('full-revisionid')
 del get_versions, v
 
+
+# GH 27101
+# TODO: remove Panel compat in 1.0
+if pandas.compat.PY37:
+    def __getattr__(name):
+        if name == 'Panel':
+            import warnings
+            warnings.warn(
+                "The Panel class is removed from pandas. Accessing it "
+                "from the top-level namespace will also be removed in "
+                "the next version",
+                FutureWarning, stacklevel=2)
+            return None
```

Comment:
Downstream are expecting this to be a type so they can do `isinstance(thing, pd.Panel)`. Will make another dummy `class Panel` here (not outside the getattr, so that we don't pollute the namespace on py37+).